benches/src/db_lookup_times_utils/seed.rs

```diff
@@ -0,0 +1,193 @@
+        block::{
+            Block,
+            PartialFuelBlock,
+        },
+        header::{
+            ConsensusHeader,
+            PartialBlockHeader,
+        },
+        primitives::Empty,
+    },
+    fuel_tx::{
+        Bytes32,
+        Transaction,
+        UniqueIdentifier,
+    },
+    fuel_types::BlockHeight,
+};
+use itertools::Itertools;
+
+fn seed_matrix<SeedClosure, Description>(
+    method: LookupMethod,
+    seed_closure: SeedClosure,
+) -> DbLookupBenchResult<impl FnOnce()>
+where
+    SeedClosure:
+        Fn(&mut RocksDb<Description>, BlockHeight, u32) -> DbLookupBenchResult<()>,
+    Description: DatabaseDescription,
+{
+    for (block_count, tx_count) in matrix() {
+        let mut database = open_rocks_db(block_count, tx_count, method)?;
+        seed_closure(&mut database, block_count, tx_count)?;
+    }
+
+    Ok(move || std::fs::remove_dir_all(get_base_path_from_method(method)).unwrap())
+}
+
+pub fn seed_compressed_blocks_and_transactions_matrix(
+    method: LookupMethod,
+) -> DbLookupBenchResult<impl FnOnce()> {
+    seed_matrix(method, |database, block_count, tx_count| {
+        seed_compressed_blocks_and_transactions(database, block_count, tx_count)
+    })
+}
+
+pub fn seed_full_block_matrix() -> DbLookupBenchResult<impl FnOnce()> {
+    seed_matrix(
+        LookupMethod::FullBlock,
+        |database, block_count, tx_count| {
+            seed_full_blocks(database, block_count, tx_count)
+        },
+    )
+}
+
+fn generate_bench_block(
+    height: BlockHeight,
+    tx_count: u32,
+) -> DbLookupBenchResult<Block> {
+    let header = PartialBlockHeader {
+        application: Default::default(),
+        consensus: ConsensusHeader::<Empty> {
+            height,
+            ..Default::default()
+        },
+    };
+    let txs = generate_bench_transactions(tx_count);
+    let block = PartialFuelBlock::new(header, txs);
+    block
+        .generate(&[], Default::default())
+        .map_err(|err| anyhow!(err))
+}
+
+fn generate_bench_transactions(tx_count: u32) -> Vec<Transaction> {
+    let mut txs = Vec::with_capacity(tx_count as usize);
+    for _ in 0..tx_count {
+        txs.push(Transaction::default_test_tx());
+    }
+    txs
+}
+
+fn height_key(block_height: BlockHeight) -> Vec<u8> {
+    block_height.to_bytes().to_vec()
+}
+
+pub fn insert_compressed_block(
+    database: &mut RocksDb<BenchDatabase>,
+    height: BlockHeight,
+    tx_count: u32,
+) -> DbLookupBenchResult<Block> {
+    let block = generate_bench_block(height, tx_count)?;
+
+    let compressed_block = block.compress(&chain_id());
+    let height_key = height_key(height);
+
+    let raw_compressed_block = postcard::to_allocvec(&compressed_block)?.to_vec();
+    let raw_transactions: Vec<(Bytes32, Vec<u8>)> = block
+        .transactions()
+        .iter()
+        .map(|tx| -> DbLookupBenchResult<(Bytes32, Vec<u8>)> {
+            let tx_id = tx.id(&chain_id());
+            let raw_tx = postcard::to_allocvec(tx)?.to_vec();
```

Comment:
Since you already defined your database, I think you can just define 3 tables as well and use storage traits to serialize and deserialize values. Plus, you can create `InMemoryTransaction` and push changed at once. It should speed up seeding of the database